src/datasources/__tests__/graphDb.test.js

```diff
@@ -1556,34 +1556,99 @@ describe('graphDb', () => {
-        .post('/', (body) => {
-          const parsedBody = typeof body === 'string' ? JSON.parse(body) : body
-          const gremlinQuery = parsedBody?.gremlin || ''
+    describe('when the user is in no groups', () => {
+      test('the response contains no related collections', async () => {
+        nock(/example-graphdb/)
+          .post('/', (body) => {
+            const parsedBody = typeof body === 'string' ? JSON.parse(body) : body
+            const gremlinQuery = parsedBody?.gremlin || ''
 
-          const hasCorrectPermittedGroups = gremlinQuery.includes('within(\'registered\',\'guest\')')
-          const hasOtherV = gremlinQuery.includes('.otherV()')
+            const hasCorrectPermittedGroups = gremlinQuery.includes('within(\'registered\',\'guest\')')
+            const hasOtherV = gremlinQuery.includes('.otherV()')
 
-          return hasCorrectPermittedGroups && hasOtherV
+            return hasCorrectPermittedGroups && hasOtherV
+          })
+          .reply(200, relatedCollectionsResponseEmptyMocks)
+
+        nock(/example-urs/)
+          .get(/groups_for_user/)
+          .reply(200, {})
+
+        const response = await graphDbDatasource(
+          { conceptId: 'C100000-EDSC' },
+          {
+            limit: 1
+          },
+          {
+            headers: {
+              'Client-Id': 'eed-test-graphql',
+              'CMR-Request-Id': 'abcd-1234-efgh-5678'
+            },
+            edlUsername: 'someEdlUsername'
+          },
+          parsedInfo
+        )
+        expect(response).toEqual({
+          count: 0,
+          items: []
         })
-        .reply(200, relatedCollectionsResponseEmptyMocks)
+      })
+    })
+  })
 
-      nock(/example-urs/)
-        .get(/groups_for_user/)
-        .reply(200, {})
+  describe('When runGraphdb is false', () => {
+    beforeEach(() => {
+      parsedInfo = {
+        name: 'relatedCollections',
+        alias: 'relatedCollections',
+        args: {
+          limit: 1
+        },
+        fieldsByTypeName: {
+          RelatedCollectionsList: {
+            count: {
+              name: 'count',
+              alias: 'count',
+              args: {},
+              fieldsByTypeName: {}
+            },
+            items: {
+              name: 'items',
+              alias: 'items',
+              args: {},
+              fieldsByTypeName: {
+                RelatedCollection: {
+                  id: {
+                    name: 'id',
+                    alias: 'id',
+                    args: {},
+                    fieldsByTypeName: {}
+                  },
+                  title: {
+                    name: 'title',
+                    alias: 'title',
+                    args: {},
+                    fieldsByTypeName: {}
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    })
 
+    test('returns an empty result and does not call the network when runGraphdb is false', async () => {
```

Comment:
The absence of a nock is valid, but not the easiest to understand. You should assert that the graphdb function is not called